Linking issue with eccodes 2.19.1

<!--
Thanks for reporting your issue.
Please fill out the sections below.
-->

- [ ] I read [the conda-forge documentation](https://conda-forge.org/docs/user/introduction.html#how-can-i-install-packages-from-conda-forge) and could not find the solution for my problem there.

Issue:

See https://github.com/conda-forge/eccodes-feedstock/issues/120 for all the details.

Something has changed in the conda-forge ecosystem in the last 24 hours that is causing some solvers to choose eccodes 2.19.1 when installing python-eccodes. Before that eccodes 2.20.0 was being used and works fine. However, with 2.19.1 I get:

```
>       from ._bindings import ffi, lib
E       ImportError: /usr/share/miniconda3/envs/test-environment/lib/python3.8/site-packages/gribapi/_bindings.cpython-38-x86_64-linux-gnu.so: undefined symbol: codes_bufr_key_is_header

/usr/share/miniconda3/envs/test-environment/lib/python3.8/site-packages/gribapi/bindings.py:29: ImportError
```

Forcing the eccodes version to 2.20.0 or above seems to have fixed things.

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -10,7 +10,7 @@ source:
   sha256: 7251436ddc20bded9a96408c705360d282918f2a6003dee05457b62d962b8491
 
 build:
-  number: 0
+  number: 1
   skip: true  # [py2k]
 
 requirements:

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -1,5 +1,5 @@
 {% set version = "2021.03.0" %}
-{% set eccodes_version = "2.16" %}
+{% set eccodes_version = "2.20.0" %}
 
 package:
   name: python-eccodes

